src/main/java/com/jcraft/jsch/Session.java

```diff
@@ -883,9 +885,20 @@ private void checkHost(String chost, int port, KeyExchange kex) throws JSchExcep
       }
       // System.err.println("finger-print: "+key_fprint);
       if (userinfo != null) {
+		if (shkc.equals("ask")) {
         boolean foo = userinfo.promptYesNo("The authenticity of host '" + chost
             + "' can't be established.\n" + key_type + " key fingerprint is " + key_fprint + ".\n"
             + "Are you sure you want to continue connecting?");
+		} else {
+		boolean foo = false; 
+    	  for(String eachFingerPrint : fingerprintList){
+    		  if(key_fprint.equalsIgnoreCase(eachFingerPrint))
+    		  {
+    			  foo = true;
+    			  break;
+    		  }
+    	  }	
+		}
```

Comment:
Also, can you point at any other applications or libraries that perform SSH host key verification using fingerprints?
This seems like a pretty unusual (and potentially insecure) practice that I am unaware of, and I'm not sure we want to directly enable this sort of thing natively within JSch.

Reply:
I have no idea if any other application which is using Jsch library is also using fingerprints to perform SSH host key verification. It is being used in SAP PI SFTP Adapter.